src/pages/ConvertVrt/Convert/index.tsx

```diff
@@ -1,141 +1,218 @@
 /** @jsxImportSource @emotion/react */
-import React from 'react';
+import React, { useCallback } from 'react';
 import BigNumber from 'bignumber.js';
 import noop from 'noop-ts';
 import { Typography } from '@mui/material';
 import {
   FormikTokenTextField,
-  LabeledProgressBar,
   FormikSubmitButton,
   ConnectWallet,
+  EnableToken,
   Icon,
+  Token,
   TokenTextField,
 } from 'components';
+import useSuccessfulTransactionModal from 'hooks/useSuccessfulTransactionModal';
+import toast from 'components/Basic/Toast';
 import { useTranslation } from 'translation';
 import useConvertToReadableCoinString from 'hooks/useConvertToReadableCoinString';
-import { format } from 'utilities/common';
-import { formatI18nextRelativetimeValues } from 'utilities';
+import PLACEHOLDER_KEY from 'constants/placeholderKey';
 import { AmountForm, ErrorCode } from 'containers/AmountForm';
+import { getContractAddress, formatI18nextRelativetimeValues } from 'utilities';
+import { InternalError } from 'utilities/errors';
+import { convertCoinsToWei, convertWeiToCoins } from 'utilities/common';
 import { VRT_ID, XVS_ID, VRT_DECIMAL } from '../constants';
 import { useStyles } from '../styles';
 
-interface ConvertProps {
-  xvsTotalWei: BigNumber;
-  xvsToVrtRate: BigNumber;
-  vrtLimitUsedWei: BigNumber;
-  vrtLimitWei: BigNumber;
-  vrtConversionEndTime: string | undefined;
+export interface IConvertProps {
+  xvsToVrtConversionRatio: BigNumber | undefined;
+  vrtConversionEndTime: Date | undefined;
+  userVrtBalanceWei: BigNumber | undefined;
+  vrtConversionLoading: boolean;
+  userVrtEnabled: boolean;
+  convertVrt: (amount: string) => Promise<string>;
   walletConnected: boolean;
 }
 
-const Convert: React.FC<ConvertProps> = ({
-  xvsTotalWei,
-  xvsToVrtRate,
-  vrtLimitUsedWei,
-  vrtLimitWei,
+const Convert: React.FC<IConvertProps> = ({
+  xvsToVrtConversionRatio,
   vrtConversionEndTime,
+  userVrtBalanceWei,
+  vrtConversionLoading,
+  userVrtEnabled,
+  convertVrt,
   walletConnected,
 }) => {
   const styles = useStyles();
   const { t, Trans } = useTranslation();
+  const { openSuccessfulTransactionModal } = useSuccessfulTransactionModal();
+  const vrtConverterProxyAddress = getContractAddress('vrtConverterProxy');
+
   const readableXvsAvailable = useConvertToReadableCoinString({
-    valueWei: xvsTotalWei,
+    valueWei: userVrtBalanceWei?.times(xvsToVrtConversionRatio || 1),
```

Comment:
I don't think we should default the conversation ratio to 1, as this could lead to displaying incorrect data. We should display the placeholder key if xvsToVrtConversionRatio is undefined.